File: view/lib/src/app/fifo.dart
import 'package:view/src/model/process.dart';
import 'package:view/src/app/aux.dart' show Aux;

Map<int, List<int>> fifo(List<Process> processList) {
  Aux aux = Aux();
  List<Process> sortedProcessList = aux.quickSortProcessByInitTime(processList);

  List<int> listDone = [];
  List<int> executionList = [];
  int n = sortedProcessList.length;

  int time = 0;
  while (listDone.length != n) {
    List<Process> l = sortedProcessList
        .where((p) => p.getTimeInit() <= time && !listDone.contains(p.getId()))
        .toList();

    if (l.isNotEmpty) {
      Process p = l.first;
      for (int i = 0; i < p.getTtf(); i++) {
        executionList.add(p.getId());
      }
      time += p.getTtf();
      listDone.add(p.getId());
    } else {
      time++;
      executionList.add(-3);
    }
  }

  // print("executionList: $executionList");
  return aux.listToMatrix(executionList);
}

File: view/lib/src/app/fifo.py
def _quick_sort_process_by_init_time(process_list):
    if(len(process_list) <= 1):
        return process_list
    else:
        pivot_time = process_list[len(process_list) // 2].getTimeInit()
        left = [p for p in process_list if p.getTimeInit() < pivot_time]
        middle = [p for p in process_list if p.getTimeInit() == pivot_time]
        right = [p for p in process_list if p.getTimeInit() > pivot_time]

        return _quick_sort_process_by_init_time(left) + middle + _quick_sort_process_by_init_time(right)

def fifo(process_list):
    sorted_process_list = _quick_sort_process_by_init_time(process_list)
    return sorted_process_list
